src/librustc_metadata/creader.rs

```diff
@@ -853,6 +853,33 @@ impl<'a> CrateLoader<'a> {
         }
     }
 
+    fn inject_profiler_runtime(&mut self) {
+        if self.sess.opts.debugging_opts.profile {
+            let mut uses_std = false;
+            self.cstore.iter_crate_data(|_, data| {
```

Comment:
I orignially copied that over from compiler_builtins. I think it's useful because a no_std crate may want to supply their own profiler builtins, e.g. to submit the results somewhere in a semihosting implementation.